src/main/java/com/umbr3114/controllers/CommentController.java

```diff
@@ -0,0 +1,160 @@
+        String postId = paramHelper.valueOf("postId");
+        String startAfterId = paramHelper.valueOf("startAfterId");
+        String date = paramHelper.valueOf("date");
+        PostModel post = mongoCollectionPost.findOne(eq("_id", new ObjectId(postId)));
+        List<CommentModel> commentModelList;
+        if(post == null){
+            return "post DNE";
+        }
+        if(startAfterId == null){
+            if(date == null){
+                Date thedate = new java.util.Date(System.currentTimeMillis());
+                DateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy");
+                date = dateFormat.format(thedate);
+            }
+            commentModelList = getByDate(postId, date);
+        }else{
+            commentModelList = getAfterId(postId, startAfterId);
+        }
+        if(postId == null){
+            return "not ok";
+        }
+        CommentListingModel commentList = new CommentListingModel();
+        commentList.count = commentModelList.size();
+        if (commentModelList.size() > 0){
+            commentList.lastId = commentModelList.get(commentModelList.size()-1)._id.toString();
+            commentList.postId = commentModelList.get(0).getPostIdString();
+        }
+        commentList.comments = commentModelList;
+        return commentList;
+
+    });
+
+    private static List<CommentModel> getAfterId(String postId, String startAfterId){
+        JacksonMongoCollection<CommentModel> mongoCollectionComments =
+                new CollectionFactory<CommentModel>(ServiceLocator.getService().dbService(), CommentModel.class).getCollection();
+        JacksonMongoCollection<PostModel> mongoCollectionPost =
+                new CollectionFactory<PostModel>(ServiceLocator.getService().dbService(), PostModel.class).getCollection();
+        FindIterable<CommentModel> iterable;
+        PostModel post = mongoCollectionPost.findOne(Filters.eq("_id", postId));
+        List<CommentModel> commentModels = new ArrayList<>();
+        if(post == null){
+            return commentModels;
+        }else{
+            iterable = mongoCollectionComments.find(Filters.and(Filters.eq("post_id", postId), Filters.gt("_id", startAfterId.toString())));
+        }
+        iterable.limit(10);
+        iterable.into(commentModels);
+        return commentModels;
+    }
+
+    private static List<CommentModel> getByDate(String postId, String date){
+        JacksonMongoCollection<CommentModel> mongoCollectionComments =
+                new CollectionFactory<CommentModel>(ServiceLocator.getService().dbService(), CommentModel.class).getCollection();
+        FindIterable<CommentModel> iterable;
+        List<CommentModel> commentList = new ArrayList<>();
+        iterable = mongoCollectionComments.find(Filters.and(eq("dateString", date), Filters.eq("post_id", postId))).limit(10);
+        iterable.into(commentList);
+        return commentList;
+    }
+
```

Comment:
fetching by date is not required. generally this method is pretty good. Only two things I could say is use a variable for a fetch limit (that way the limit can easily be changed or controlled). Also, don't pass a string for the date to the database, but an actual date object. this is absolutely not a requirement, since passing a String will definitely work, but passing a date object causes it to be converted into seconds, which is faster for the database to compare (will cause this route to perform faster).

other than that, this function is great.